activerecord/lib/active_record/associations/has_many_association.rb

```diff
@@ -117,8 +115,25 @@ def delete_or_nullify_all_records(method)
 
         # Deletes the records according to the <tt>:dependent</tt> option.
         def delete_records(records, method)
-          if method == :destroy
+          case method
+          when :destroy
             records.each(&:destroy!)
+            update_counter(-records.length) unless reflection.inverse_updates_counter_cache?
+          when :destroy_async
+            association_class = records.first.class
+            primary_key_column = association_class.primary_key.to_sym
+            ids = records.collect { |assoc| assoc.public_send(primary_key_column) }
+
+            enqueue_destroy_association(
+              owner_model_name: owner.class.to_s,
+              owner_id: owner.id,
```

Comment:
`owner.id` is already used somewhere else (i.e., `activerecord/lib/active_record/associations/has_many_association.rb`). What if the owner does not have an `id` because of the custom primary key? Is it safe to use it here?

Reply:
I assume the most correct thing would be `owner.send(owner.class.primary_key)`

Reply:
Actually, no, `owner.id` is fine: https://github.com/rails/rails/blob/f95c0b7e96eb36bc3efc0c5beffbb9e84ea664e4/activerecord/lib/active_record/attribute_methods/primary_key.rb#L19